### ISSUE TITLE
Add eBPF support for IPv6

### ISSUE THREAD

**Issue by telmich** [projectcalico/calico#4736]
## Expected Behavior

* calico support eBPF mode for IPv6 only clusters

## Current Behavior

* it is not supported

## Context

* Cannot use kube-proxy free environments in IPv6 only custers

## Your Environment
* Calico version: v3.19.0
* Orchestrator version (e.g. kubernetes, mesos, rkt): kuberentes
* Operating System and version: Alpine Linux 3.14
* Link to your project (optional): www.datacenterlight.ch

**Comment by lwr20** [comment]
@fasaxc FYI

**Comment by mazdakn** [comment]
@svallala we have started the development, but cannot provide a timeline at the moment.

**Comment by MTRNord** [comment]
Hi any updates for this in 2023? :) (A "no update" is still an update I would appreciate. :))

**Comment by tomastigera** [comment]
We did not get much further due to other business priorities, and also other dev priorites as we needed to restructure the dp to be able to add some features and also ipv6 in a simpler way. We had to take a step back. That should be ready for 3.27 and my goal is to resume ipv6 as the very next thing :crossed_fingers:

**Comment by MTRNord** [comment]
> We did not get much further due to other business priorities, and also other dev priorites as we needed to restructure the dp to be able to add some features and also ipv6 in a simpler way. We had to take a step back. That should be ready for 3.27 and my goal is to resume ipv6 as the very next thing crossed_fingers

thanks a lot for this update :)

**Comment by telmich** [comment]
@tomastigera in case there is something to support you with such as an ipv6 only k8s cluster, let me know. 

Being able to use eBPF and thus allow source IP preservation would be highly appreciated in the ipv6 context, as services can be directly accessed from the Internet. And for many services, being able receive the original IP address is crucial.

**Comment by davidwadddell** [comment]
hi guys appreciate the work on calico.  wondered if support for this was any closer. source IP preservation is a desirable feature for us but is blocked by lack of ip6  support with eBPF; support is present in Cilium for this at the moment, but we'd prefer not to switch,

**Comment by tomastigera** [comment]
We are actually making a good progress and we started passing fv tests with IPv6 code. With the summer vacations in full swing, my personal estimate is that we may have a beta of IPv6 released around end of September :crossed_fingers:

**Comment by davidwadddell** [comment]
Thanks for the update @tomastigera

**Comment by tomastigera** [comment]
I am closing this issue as DONE as ipv6 support is merged into master and will be released as in 3.27.0. There are many limitations like no support for dual-stack xdp etc. Please refer to docs. Any follow up ipv6 issues should be reported and tracked as separate issue.
